packages/dx-grid-core/src/plugins/local-selection/computeds.js

```diff
@@ -0,0 +1,20 @@
+export const getAvailableToSelect = (rows, getRowId, isGroupRow) => {
+  let dataRows = rows;
+  if (isGroupRow) {
+    dataRows = dataRows.filter(row => !isGroupRow(row));
+  }
+  return dataRows.map(row => getRowId(row));
+};
+
+export const isSomeSelected = ({ selection, availableToSelect }) => {
+  const selectionRows = new Set(selection);
```

Comment:
Just a thought. Is it possible to convert the selection array to set once?

Reply:
- We can setState({selectionRows: new Set(selection)}) on "rows" getter. But I think what double call `new Set()` is faster that once `setState()`.
- If we make closure with `new Set(selection)`. Maybe we can give an old selection from closure because getters allSelected and someSelected don't always call together. I think

Reply:
Ok, lets try to change selection Getter type from array to set. Like we did with expandedGroups Getter in GroupingState.